### PR TITLE
Add `get_alerts` and `get_run_id_from_name` methods to Client

### PR DESCRIPTION
**Pull request by wk9874**
Added a `get_run_id_from_name` method to the Client class. Because the API endpoints now require run ID instead of run name, it made using the Client class more difficult. By adding this endpoint, it means that users can get the run ID from the name first, before passing that to other Client class methods. Since multiple runs of the same name are now allowed (not sure if that is intentional or a bug), if the API endpoint returns multiple IDs (or no IDs) for a given run name, an exception is raised.

Added a `get_alerts` method to the Client class. This uses the `alerts` information from the `get_run` API endpoint response to provide the user with current details about alerts for a given run. Includes the following options:
- `critical_only`: Whether to only return details about alerts which are currently critical. Very useful for adding to a script to catch alerts as they become critical and use as a feedback to control the run itself (eg if an alert of a given name is contained in the returned list, stop execution of the run since it is pointless to continue)
- `names_only`: Whether to only return the names of the alerts, or the full details about them. Most of the time only the name may be needed, as in the example given above.

Added test for `get_alerts()` method

### CODE REVIEW

File: simvue/client.py [hunk 1]
```diff
@@ -468,3 +484,36 @@ def get_events(self, run, filter=None, start=0, num=0):
...
+            Whether to only return details about alerts which are currently critical, by default True
+        names_only: bool, optional
+            Whether to only return the names of the alerts (otherwise return the full details of the alerts), by default True
+        """
+        response = requests.get(f"{self._url}/api/runs/{run}", headers=self._headers)
+
+        if response.status_code == 404:
+            if 'detail' in response.json():
+                if response.json()['detail'] == 'run does not exist':
+                    raise Exception('Run does not exist')
```
Comment (kzscisoft) [inline simvue/client.py:505]: Please avoid raising raw Exceptions (I know Simvue has this in places - I have some changes in to remove them).

I recommend `RuntimeError` here 🙂

File: simvue/client.py [hunk 2]
```diff
@@ -468,3 +484,36 @@ def get_events(self, run, filter=None, start=0, num=0):
...
+        critical_only : bool, optional
+            Whether to only return details about alerts which are currently critical, by default True
+        names_only: bool, optional
+            Whether to only return the names of the alerts (otherwise return the full details of the alerts), by default True
+        """
+        response = requests.get(f"{self._url}/api/runs/{run}", headers=self._headers)
+
+        if response.status_code == 404:
+            if 'detail' in response.json():
+                if response.json()['detail'] == 'run does not exist':
```
Comment (kzscisoft) [inline simvue/client.py:504]: There is a lot of potential for `KeyError` issues with accessing the JSON response via nested keys, would be good to handle them gracefully.

**Review by kzscisoft — CHANGES_REQUESTED** [review]
See comments

File: simvue/client.py (hunk 2 above)
Comment (wk9874) [reply simvue/client.py:504]: This matches how all other Client methods perform this check - I would say that we keep as is for this MR, and then could add a new MR to handle key errors more gracefully across all Client methods at a later point

**Pull request merged**